### PR TITLE
proxmark3: Replace with rfidresearchgroup fork

### PR DESCRIPTION
**Pull request by nyanotech**
Looking at [repology](https://repology.org/project/proxmark3/versions), it seems like all the other distros that have proxmark3 packaged package the fork version. Additionally, the upstream version hasn't been updated in some time.

Debating if I should add a release note for this?

###### Description of changes

* Replaces `proxmark3` with what used to be `proxmark3-rrg`
* Aliases `proxmark3-rrg` to `proxmark3`
* Removes `proxmark3-unstable`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by emilytrau — APPROVED** [review]
I think this is a good idea 👍

File: pkgs/tools/security/proxmark3/default.nix [hunk 1]
```diff
@@ -1,61 +1,86 @@
...
+, withGeneric ? false
+, withSmall ? false
+, withoutFunctions ? []
+, hardwarePlatform ? if withGeneric then "PM3GENERIC" else "PM3RDV4"
+, hardwarePlatformExtras ? lib.optionalString withBlueshark "BTADDON"
+, standalone ? "LF_SAMYRUN"
+}:
+assert withBlueshark -> stdenv.hostPlatform.isLinux;
+stdenv.mkDerivation rec {
+  pname = "proxmark3-rrg";
```
Comment (n0emis) [inline pkgs/tools/security/proxmark3/default.nix:26]:
```suggestion
  pname = "proxmark3";
```

**Review by n0emis — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/proxmark3/default.nix (hunk 1 above)
Comment (nyanotech) [reply pkgs/tools/security/proxmark3/default.nix:26]: >_<

fixed

**Review by n0emis — APPROVED** [review]

**Review by fpletz — APPROVED** [review]
Thank you!

**Pull request merged**